encoding/kzg/prover/prover.go

```diff
@@ -12,18 +12,18 @@ import (
 	"sync"
 
 	"github.com/Layr-Labs/eigenda/encoding"
-	"github.com/Layr-Labs/eigenda/encoding/kzgrs"
+	"github.com/Layr-Labs/eigenda/encoding/fft"
+	"github.com/Layr-Labs/eigenda/encoding/kzg"
 	"github.com/Layr-Labs/eigenda/encoding/rs"
-	kzg "github.com/Layr-Labs/eigenda/pkg/kzg"
+	"github.com/consensys/gnark-crypto/ecc/bn254"
 
-	bls "github.com/Layr-Labs/eigenda/pkg/kzg/bn254"
 	_ "go.uber.org/automaxprocs"
 )
 
 type Prover struct {
-	*kzgrs.KzgConfig
+	*kzg.KzgConfig
 	Srs          *kzg.SRS
-	G2Trailing   []bls.G2Point
+	G2Trailing   []bn254.G2Affine
 	mu           sync.Mutex
 	LoadG2Points bool
 
```

Comment:
Compared to the `Prover -> ParameterizedProver`, what do you think `ProverFactory -> Prover`?

Reply:
I don't prefer ProverFactory, because the generated prover share common object with "factory", like srs

Reply:
It's totally ok to share things across the objects created from the factory. SRS is even less a concern since it's just immutable data resource.

Reply:
(just for an example: https://github.com/cockroachdb/cockroach/blob/c9c3cc5f3c3a4a6ab556f4b9d5b6ec0381901bdb/pkg/sql/catalog/descs/factory.go#L141)

Reply:
@jianoaix, the issue here is that the `Prover` struct rather than the `ParametrizedProver` is implementing the `Prover` interface, which is the way it has to be, since the prover interface accepts arbitrary pararms. 

It would be strange to have a `ProverFactory` implement the `Prover` interface. So I think your issue may be with the interfaces.

Reply:
ProverFactory doesn't need to tie its interface to Prover, they can be completely different/separate (and usually FooFactory has different interface than Foo).
Separating this issue out sounds fine to me.